src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -303,42 +327,43 @@ internal static TraceLevel GetFunctionTraceLevel(IFunctionInstance functionInsta
         }
 
         /// <summary>
-        /// If the specified function instance requires a timeout (either via <see cref="TimeoutAttribute"/>
-        /// or because <see cref="JobHostConfiguration.FunctionTimeout"/> has been set, create and start the
-        /// timer.
+        /// If the specified function instance requires a timeout (via <see cref="TimeoutAttribute"/>),
+        /// create and start the timer.
         /// </summary>
         [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope")]
-        internal static System.Timers.Timer StartFunctionTimeout(IFunctionInstance instance, TimeSpan? globalTimeout, CancellationTokenSource cancellationTokenSource, TraceWriter trace)
+        internal static System.Timers.Timer StartFunctionTimeout(IFunctionInstance instance, TimeoutAttribute attribute, CancellationTokenSource cancellationTokenSource, TraceWriter trace, Action onElapsedCallback)
         {
-            MethodInfo method = instance.FunctionDescriptor.Method;
-            if (!method.GetParameters().Any(p => p.ParameterType == typeof(CancellationToken)))
-            {
-                // function doesn't bind to the CancellationToken, so no point in setting
-                // up the cancellation timer
-                return null;
-            }
-
-            // first see if there is a Timeout applied to the method or class
-            TimeSpan? timeout = globalTimeout;
-            TimeoutAttribute timeoutAttribute = TypeUtility.GetHierarchicalAttributeOrNull<TimeoutAttribute>(method);
-            if (timeoutAttribute != null)
+            TimeSpan? timeout = null;
+            if (attribute != null)
```

Comment:
Just to make things really clear, I'd put an if (attribute == null) { return null } first thing in the method.